src/en/developer-resources.md

```diff
@@ -0,0 +1,131 @@
+that controller can access the resources, or the Juju Charm Store where access
+is controlled by permissions.
+
+## Listing resources
+
+### juju list-resources
+
+Users can list the resources that are currently available on the Juju
+controller by using the `juju list-resources` command. The command shows
+resources for a service or a unit.
+
+```sh
+$ juju list-resources resources-example
+[Service]
+RESOURCE SUPPLIED BY REVISION
+software admin@local 2016-25-05T18:37
+
+$ juju list-resources resources-example/0
+[Unit]
+RESOURCE REVISION
+software 2016-25-05T18:37
+```
+
+### charm list-resources
+
+Users can display the resources tat are currently available in Juju Charm Store
+for a charm or a specific revision number with the `charm list-resources`
+command.
+
+```sh
+$ charm list-resources cs:~lazypower/etcd
+[Service]
+RESOURCE REVISION
+etcd     0
+etcdctl  0
+```
+
+## Adding resources
+
+### juju attach
+
+The `juju attach` command uploads a file from local disk to the Juju controller
+to be used as a resource for a service. You must specify the charm name, the
+resource name and the path to the file.
+
+```sh
+juju attach charm-name resource-name=filepath
+```
+
+### juju deploy
+
+Resources may be uploaded to the Juju controller at deploy time by specifying
+the --resource flag followed by resource-name=filepath pair. This flag may be
+repeated more than once to upload more than one resource.
+
+```sh
+juju deploy charm-name --resource foo=/some/file.tgz --resource bar=./docs/cfg.xml
+```
+Where "foo" and "bar" are the resource names in `metadata.yaml` file for the
+charm-name charm.
+
+### charm attach
+
+The `charm attach` command uploads a file to the Juju Charm Store as a new
+resource for the charm.
+
+```sh
+charm attach ~mbruzek/trusty/consul software=./consul_0.6.4_linux_amd64.zip
+```
+
+# Using resources in a charm
+
+### resource-get
+
+The charm command `resource-get` will fetch a resource from the Juju
+controller or the Juju Charm store. The command returns a local path to the
+file for a named resource.
+
+If `resource-get` has not been run for the named resource previously, then the
+resource is downloaded from the controller at the revision associated with the
+unit's service. That file is stored in the unit's local cache. If `resource-get`
+*has* been run before then each subsequent run synchronizes the resource with
+the controller. This ensures that the revision of the unit-local copy of the
+resource matches the revision of the resource associated with the unit's
+service.
+
+The path provided by `resource-get` references the up-to-date file for the
+resource. Note that the resource may get updated on the controller for the
+service at any time, meaning the cached copy *may* be out of date at any time
+after `resource-get` is called. Consequently, the command should be run at
+every point where it is critical for the resource be up to date.
```

Comment:
This is only sort of true.  If the resource updates, then the  charm will get a charm-upgrade hook firing.  You can safely choose to ignore it during charm-update, if you also do resource-get before you need to use the resource.  Note that resource-get won't download anything if you already have the most up-to-date version of the resource, so it's safe to just always call resource-get first, since it'll just be a quick API call unless you're out of date.